compiler.js

```diff
@@ -1,3 +1,68 @@
+let examples = new Map();
+
+examples["helloWorld"] = `// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+contract MyContract {
+    function helloWorld() public pure returns (string memory) {
+        return "Hello, World!";
+    }
+}`;
+
+examples["erc20"] = `// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract ERC20 {
+    string public constant name = "ERC20";
+    string public constant symbol = "ERC";
+    uint8 public constant decimals = 18;
+
+    mapping(address => uint256) balances;
+    mapping(address => mapping (address => uint256)) allowed;
+
+    event Transfer(address indexed from, address indexed to, uint256 tokens);
+    event Approval(address indexed owner, address indexed spender, uint256 tokens);
+    
+    uint256 m_totalSupply;
```

Comment:
Make this a public immutable and remove the `totalSupply` function?

Reply:
Totally makes sense